### PR TITLE
Radio Group sketch library

### PR DESCRIPTION
**Pull request by dhastwell**
This is for the new sketch library for "LeafyGreen-ified" radio group component. Open for review and feedback.

Requested areas for review:
- Consistency
- Resizing of components
- Nesting, symboling, locking and layer naming
- Light, Dark and Radio box options
- General comments

### CODE REVIEW

**Review by audroue — COMMENTED** [review]
Overall looks awesome Dave. 

Here are some nits: 

- I would remove focus/hover states for the library since designers don't use these in their comps/prototypes and can add noise to the library. I would however definitely include these in the InVision Inspect for when we need to build these states out.
- There are some issues around fixing the symbol and position for the small and x-small sizes. (I can provide a higher res image on Slack if you need it.)
image

- I would prefix static with “_” so that it appears first on the list.
image


- I would consider allowing for only text overrides, in my experience people usually don’t know how to use or get confused with overrides that are more complex than just text + I would rename the text layer containing “I agree to this thin” to something like “Radio Input Copy”.
image

- Would you be open to renaming the file and cover from “radio group” to “radio inputs”?

**Comment by dhastwell** [comment]
Thanks for the feedback, Rob. I agree, the hover states probably adds noise. Myself, I tend to use focus states in my designs, as they can be a fundamental specification to a story for the engineers or a screen that impacts the user experience. I cannot say I have found hover states useful though.

I am not 100% sure if we want to keep in the component? As a designer on Charts, as long as the LeafyGreen built component is flexible to override the font-size, I wouldn't advocate for an X-Small size in the sketch library if we are moving away from that direction. (Slightly ironic because I made the X-small size and included it in the library) Happy to take it out if we decide to do that.

I will action those other points but I might wait till @DesignerDave to add his input.

**Comment by audroue** [comment]
Sure thing. Agreed focus states are good to have.

**Comment by DesignerDave** [comment]
Looks like my license for Sketch just ended, so I'm getting set up with Sketch teams. Sorry in advance for the delayed review.

Generally, focus and hover states are needed for design specs for our implementation of the component, but it's okay to omit those in the Sketch library for the team! Since we'll build (or in this case, update) the component, the focus state will be baked in, and thus the resource developers will be using will have that focus state out-the-box. Given that, it's probably just added noise in the library (but once again, the specs we will be implementing needs these).

**Comment by dhastwell** [comment]
@DesignerDave Fwiw, I still have the seperate (invision link removed) with all the visible states.

**Comment by DesignerDave** [comment]
Hmm... I just got Sketch back up-and-running, but I'm unable to open the file for some reason. What version of Sketch are you on @dhastwell?

**Comment by dhastwell** [comment]
> Hmm... I just got Sketch back up-and-running, but I'm unable to open the file for some reason. What version of Sketch are you on @dhastwell?

v.66.1

**Comment by DesignerDave** [comment]
Got the library up-and-running after updating Sketch! Apologies for the delay.

*Some notes:*
- These overall look excellent, and feel like they belong in LeafyGreen!
- I LOVE the dark blue border between the input and focus state in the dark mode version. It does a great job separating the interaction ring from the input.
- I'm not sure that I understand the reason for including the extra-small version. From what I can tell, there's actually no vertical space savings for having the smaller font-size here since the input itself is the same size between `small` and `extra-small`. I'd probably suggest making `normal => large` and either `small => normal` or keeping small, "small". Additionally, we should really steer away from using 12px fonts wherever possible.
- I'm still in the camp that we should be treating these as design specifications, not engineering ones. Design Systems will need the engineering specifications, but once we build these, including focus and hover states is unnecessary. The React component will have the hover and focus states built-in by default.
- A small thing, but I'd omit the shadow on the circle in the middle of the selected input for disabled states. That just helps drive home that the input is disabled a little.
- Any thoughts around animations for the input (like we have with [checkbox](https://mongodb.github.io/leafygreen-ui/?path=/story/checkbox--default))?
- The background colors of some of the active states in dark mode are different: 
Screen Shot 2020-06-15 at 11 35 39 AM

Overall, these look like they'll be an excellent addition.

**Comment by dhastwell** [comment]
Thanks for taking a look @DesignerDave.

> including focus and hover states is unnecessary...

If you read my first comment, I did mention that I use focus states in my designs for Charts features. These have become useful for design specifications to demonstrate the flow to stakeholders and engineers and ultimately, to ensure the user experience isn't compromised. So I have to disagree here. As for hover states, I cannot say they have been useful in my experience. Though I have used them sporadically in my designs to showcase Leafygreen to the wider team so they can see how beautiful Leafygreen is. ;)

> I'm not sure that I understand the reason for including the extra-small version...

As I mentioned in our last catch-up, the current chart builder UI has a lot of size 10, 11, 12px font sizes and we will soon be upping all those to 12px. As long as the component is built to be flexible to cater for our teams needs, I'd be happy to just keep it to two sizes and encourage no less than 14px font size where possible.

> small and extra-small. I'd probably suggest making normal => large and either small => normal or keeping small, "small".

I have tried to keep consistent with the other component sizes so that the other designers don't need to work out what size is suppose to go with what. @audroue do you have a preference here?

Again, appreciate the feedback. I am currently working on an animation using Principle.

**Comment by dhastwell** [comment]
Radio-Input
@DesignerDave @audroue @fredtruman

**Comment by dhastwell** [comment]
Forgot to mention - I left the sizes as "Normal" and "Small". Maybe designers should be considering to use the Radio Box Group if they are looking at a large size?

**Comment by audroue** [comment]
Yeah the large radio input is pretty dang large. What are your thoughts do you want to keep it? Are there use cases where we need it to be that big?

**Comment by dhastwell** [comment]
> Yeah the large radio input is pretty dang large. What are your thoughts do you want to keep it? Are there use cases where we need it to be that big?

There is no Large size @audroue. Or are you referring to the Normal? Here are screens of it in use in my recent designs (invision link removed) Sorry - Just a little confused and want to clarify what you mean.

**Comment by DesignerDave** [comment]
Hey @dhastwell! Sorry for the delay – my inbox has started to become harder to manage as of late, and I'm trying to unbury many threads 😬

> If you read my first comment, I did mention that I use focus states in my designs for Charts features. These have become useful for design specifications to demonstrate the flow to stakeholders and engineers and ultimately, to ensure the user experience isn't compromised. So I have to disagree here. As for hover states, I cannot say they have been useful in my experience. Though I have used them sporadically in my designs to showcase Leafygreen to the wider team so they can see how beautiful Leafygreen is. ;)

I may not be understanding the role of focus states in the context of LeafyGreen Sketch components. If we're building the React component ourselves, focus states are *always* included in the component by default, so it's unnecessary for the engineers on your team to even need to consider it. The goal of what we're doing is to obsolete making pixel-perfect specifications for individual components, and let designers and developers focus on the problems that are specific to the application itself. What I get from including focus states is that there's still an expectation for designers to specify to engineers what focus states should look like, when that should be cut out from the process entirely where LeafyGreen components are concerned.

Maybe you can elaborate on how you use focus states outside of providing engineering specs? It's not clear to me how they would help you demonstrate the user flow unless you're prototyping entire keyboard interaction flows.

> As I mentioned in our last catch-up, the current chart builder UI has a lot of size 10, 11, 12px font sizes and we will soon be upping all those to 12px. As long as the component is built to be flexible to cater for our teams needs, I'd be happy to just keep it to two sizes and encourage no less than 14px font size where possible.

We can make it so the size is able to be overwritten in the component for Charts purposes if that helps! I'd love to discourage smaller sizes in other applications when possible.

**Comment by dhastwell** [comment]
I understand what you are saying about obsoleting designers making pixel perfect screens to focus on the more important things @DesignerDave. Though as a designer, I want to reduce any unnecessary back and forths. To do that, I need to ensure my design specifications are on point. Ideally, I am pairing with engineers and assisting them where I can, but it is not a perfect world in reality.

Like yourself with your inbox, we are all very busy. I can confirm that the engineers use the invisions as a source of truth and can sometimes miss the UX criteria points written in the ticket (we are all human after all). So in my case, it’s handy to have this in-built into the library. Here is a good example where Jimmy recently has provided focused states in his design screens for the new fullscreen chart experience. Without this, the engineers wouldn’t know where to implement the focused state. https://mongodb.invisionapp.com/d/#/console/19973448/419391064/preview

Another potential example (one I actually supplied earlier today on ask-design-systems channel) is the focus state on the leafygreen modal. They were not from my designs or specifications, but there could have been confusion around when/when not to apply it.

**Comment by audroue** [comment]
I'm ok to move forward and integrate the buttons with the focus states.

**Comment by DesignerDave** [comment]
In the modal example, that's an issue on our end that we need to fix due to the technical implementation for modal keyboard accessibility.

In regards to my point about the radio inputs, there won't actually be a way for engineers to implement the focus state itself in the first place, so including it in the specs may be more confusing than otherwise. From a technical perspective, the Radio Input component is the sum of the text and the input, so applying styles to one but not both isn't going to be possible unless there's a good reason to specifically build a mechanism to do so.

TLDR; Engineers won't be implementing the focus states with LeafyGreen components, so adding them to design specs for usages of the LeafyGreen components will cause confusion, and not be something the engineers even have access to most of the time.

**Comment by dhastwell** [comment]
@DesignerDave @audroue I've made a couple of updates based on the discussion and feedback.

- Removed focus states in sketch library
- Renamed sketch library "Radio Group"

I have also separated the designs into it's own [invision project](https://mongodb.invisionapp.com/d/main#/projects/prototypes/20198611).

**Comment by audroue** [comment]
Hey man! Sorry for the late feedback. Hopefully this is the last round now! I know you've been working on this for a while.

Here's some stuff I noticed:

- I'd pin the smaller radio buttons to the top left, like so: 

image

so that the button doesn't float away on resize (for the smaller ones — the bigger ones work perfectly):

image

- All the other radio buttons resize perfectly except for this one:

image

Seems to be the only one that doesn't have a symbol as radio button:

image

- Final nit is I'd consider making the line-height 20px instead of the 16px which imo is a bit tight. (g and d getting pretty close.)

image

**Comment by audroue** [comment]
Hey Dave! 3 very small comments:

When I insert the reference symbol the dark background is absent. (I think the background is not part of the reference symbol artboard.)

image

Do you mind saving on the cover page, so that when we preview the file we see the cover? :) 

Also not sure we need this radio box section. If I understand these will probably not get accessed by people since they'll be using the symbols in Dark & Light. If this directory contains symbols that helped you create the file I would just call it "z_Not for Use" to indicate that people can ignore it (the z prefix makes it drop to the bottom of the list).

image

Otherwise everything else looks good to me!

**Comment by dhastwell** [comment]
@audroue Just made the requested updates now. :)

**Review by audroue — APPROVED** [review]
Looks good! Thanks for the great work! 🥳

**Review by DesignerDave — APPROVED** [review]
Cheers @dhastwell! Press the big, green button at your leisure 😄

**Comment by fredtruman** [comment]
giphy
